### PR TITLE
mago-mi - ability to force external console

### PR DESCRIPTION
**Pull request by buggins**
If --tty=external-console is set, run debuggee in new console.
MagoDE changes: LaunchSuspended pszOptions parameter == "--external-console" support is added - sets NewConsole flag for launch parameters.

### CODE REVIEW

File: DebugEngine/MagoNatDE/Engine.cpp [hunk 1]
```diff
@@ -501,6 +501,8 @@ namespace Mago
         info.StdOutput = (HANDLE) hStdOutput;
         info.StdError = (HANDLE) hStdError;
         info.Suspend = true;
+		if (pszOptions && std::wstring(pszOptions) == L"--external-console")
+			info.NewConsole = true;
```
Comment (rainers) [inline DebugEngine/MagoNatDE/Engine.cpp]: There's a mixture of tabs and spaces here. This causes bad indentation.

Wouldn't it be better to add this to the launch flags?

Comment (buggins) [inline DebugEngine/MagoNatDE/Engine.cpp]: Tabs fixed.
There is no standard flag in LAUNCH_FLAGS for external console.
But if you prefer non-standard dwLaunchFlags over non-standard pszOptions, I can use some bit as console flag.

Comment (rainers) [inline DebugEngine/MagoNatDE/Engine.cpp]: > There is no standard flag in LAUNCH_FLAGS for external console.
> But if you prefer non-standard dwLaunchFlags over non-standard pszOptions, I can use some bit as console flag.

I think it's a bit cleaner to pass it in dwLaunchFlags as there are so many possible ways to put it into a string. Just grab a bit high enough to be unlikely to be added to the MS flags.

Comment (buggins) [inline DebugEngine/MagoNatDE/Engine.cpp]: Ok. Will do so after a vacation.

Comment (rainers) [inline DebugEngine/MagoNatDE/Engine.cpp]: Thanks. Enjoy your vacation!

**Comment by buggins** [comment]
refactored external console support

File: DebugEngine/MagoNatDE/Engine.cpp [hunk 2]
```diff
@@ -501,6 +501,8 @@ namespace Mago
         info.StdOutput = (HANDLE) hStdOutput;
         info.StdError = (HANDLE) hStdError;
         info.Suspend = true;
+		if ( (dwLaunchFlags & LAUNCH_NEW_CONSOLE) != 0 )
+            info.NewConsole = true;
```
Comment (rainers) [inline DebugEngine/MagoNatDE/Engine.cpp]: There's some bad indentation here, tabs instead of spaces.

File: DebugEngine/MagoNatDE/Engine.h [hunk 3]
```diff
@@ -15,6 +15,11 @@
 #include "RemoteDebuggerProxy.h"
 #include "ExceptionTable.h"
 
+enum enum_LAUNCH_FLAGS_MAGO
```
Comment (rainers) [inline DebugEngine/MagoNatDE/Engine.h]: The `enum_` seems strange. `LAUNCH_FLAGS_MAGO` looks good enough to me.

Comment (buggins) [inline DebugEngine/MagoNatDE/Engine.h]: did not notice it

File: DebugEngine/MagoNatDE/Engine.cpp (hunk 2 above)
Comment (buggins) [inline DebugEngine/MagoNatDE/Engine.cpp]: fixed

File: DebugEngine/MagoNatDE/Engine.h (hunk 3 above)
Comment (buggins) [inline DebugEngine/MagoNatDE/Engine.h]: fixed

**Comment by rainers** [comment]
Thanks.

**Pull request merged**